src/hooks/useTransactionList.ts

```diff
@@ -85,7 +85,7 @@ export function useTransactionList(wallet: EdgeCurrencyWallet, tokenId: EdgeToke
     const cleanupChanged = wallet.on('transactionsChanged', txs => {
       let relevant = false
       for (const tx of txs) {
-        if (tx.tokenId === tokenId) {
+        if (tx.tokenId === tokenId && streamedTxs.some(streamedTx => streamedTx.txid === tx.txid)) {
```

Comment:
This is *potentially* O(n^2), but only if the core / engines have bugs that spam onTransactionChanged events. However, spamming needless events is *already* inefficient, so let's leave this as it is.

Reply:
Fixed it